src/uthash.h

```diff
@@ -142,12 +158,16 @@ do {
 #ifdef HASH_BLOOM
 #define HASH_BLOOM_BITLEN (1UL << HASH_BLOOM)
 #define HASH_BLOOM_BYTELEN (HASH_BLOOM_BITLEN/8UL) + (((HASH_BLOOM_BITLEN%8UL)!=0UL) ? 1UL : 0UL)
-#define HASH_BLOOM_MAKE(tbl)                                                     \
+#define HASH_BLOOM_MAKE(tbl,mem_ok)                                              \
 do {                                                                             \
   (tbl)->bloom_nbits = HASH_BLOOM;                                               \
   (tbl)->bloom_bv = (uint8_t*)uthash_malloc(HASH_BLOOM_BYTELEN);                 \
-  if (!(tbl)->bloom_bv) {                                                        \
-    uthash_fatal("out of memory");                                               \
+  if (!((tbl)->bloom_bv))  {                                                     \
+    if (HASH_NOMEM_OK) {                                                         \
```

Comment:
I'd still like to see this code block defined as a macro somewhere up top, and then used here without any conditional branches. Although, I also don't like the use of `break` to "skip the rest of this macro", because it's not robust to refactoring — once a macro contains a `break`, you can never pull that bit out into a shared subroutine-macro.

In other words, I'm hoping for something like this:

    #if HASH_NOMEM_OK
    #define HASH_OOM(mem_ok)  (mem_ok) = 0
    #else
    #define HASH_OOM(mem_ok) uthash_fatal("out of memory")
    #endif

    if (!(tbl)->bloom_bv) {
        HASH_OOM(mem_ok);
    } else {
        uthash_bzero((tbl)->bloom_bv, HASH_BLOOM_BYTELEN);
        (tbl)->bloom_sig = HASH_BLOOM_SIGNATURE; 
    }

but I know that might be a lot harder to pull off, with complicated things like `HASH_ADD`.